File: app/src/main/java/com/eccentricyan/pohe/common/utils/NetUtils.java
package com.eccentricyan.pohe.common.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

/**
 * Created by shiyanhui on 2017/04/15.
 */

public class NetUtils {
    public static boolean checkNet(Context context) {
        boolean mWifiConnected = isWifiConnected(context);

        boolean mMobileConnected = isMobileConnected(context);

        if (mWifiConnected == false && mMobileConnected == false) {

            return false;
        }
        return true;
    }

    private static boolean isWifiConnected(Context context) {
        ConnectivityManager manager = (ConnectivityManager) context.getSystemService(
                Context.CONNECTIVITY_SERVICE);

        NetworkInfo info = manager.getNetworkInfo(ConnectivityManager.TYPE_WIFI);
        if (info != null && info.isConnected()) {
            return true;
        }

        return false;

    }

    private static boolean isMobileConnected(Context context) {
        ConnectivityManager manager = (ConnectivityManager) context.getSystemService(
                Context.CONNECTIVITY_SERVICE);

        NetworkInfo info = manager.getNetworkInfo(ConnectivityManager.TYPE_MOBILE);
        if (info != null && info.isConnected()) {
            return true;
        }

        return false;
    }

}

File: app/src/main/java/com/eccentricyan/pohe/common/utils/NetUtils.kt
package com.eccentricyan.pohe.common.utils

import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo

/**
 * Created by shiyanhui on 2017/04/15.
 */

object NetUtils {
    fun checkNet(context: Context): Boolean {
        val mWifiConnected = isWifiConnected(context)

        val mMobileConnected = isMobileConnected(context)

        return if (mWifiConnected == false && mMobileConnected == false) {

            false
        } else true
    }

    private fun isWifiConnected(context: Context): Boolean {
        val manager = context.getSystemService(
                Context.CONNECTIVITY_SERVICE) as ConnectivityManager

        val info = manager.getNetworkInfo(ConnectivityManager.TYPE_WIFI)
        return if (info != null && info.isConnected) {
            true
        } else false

    }

    private fun isMobileConnected(context: Context): Boolean {
        val manager = context.getSystemService(
                Context.CONNECTIVITY_SERVICE) as ConnectivityManager

        val info = manager.getNetworkInfo(ConnectivityManager.TYPE_MOBILE)
        return if (info != null && info.isConnected) {
            true
        } else false

    }

}
